### PR TITLE
Weight fitting refactor

### PR DESCRIPTION
**Pull request by RasmusOrsoe**
This PR refactors the UniformWeightFitter by introducing a general purpose WeightFitter class, and also introduces a new weight fitter called "BjørnLow" that produces pr. event weights. The weights make bin counts uniform up to a user-specified `x_low`, and bin counts fall off as (1/(1 + a*(x - x_low)) for x > x_low. See attached picture.

test_energy_hist

### CODE REVIEW

File: src/graphnet/training/weight_fitting.py [hunk 1]
```diff
@@ -20,60 +22,150 @@ def __init__(
...
-        else:
-            query = f"select {self._index_column}, {variable} from {self._truth_table} where {self._index_column} in {str(tuple(selection))}"
-        with sqlite3.connect(self._database_path) as con:
-            data = pd.read_sql(query, con)
-        return data
+    def _generate_weight_name(self):
+        return self._variable + "_uniform_weight"
+
+
+class BjørnLow(WeightFitter):
```
Comment (asogaard) [inline src/graphnet/training/weight_fitting.py:121]:
```suggestion
class BjoernLow(WeightFitter):
```

File: examples/bjørn_low_weight_fitter_example.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import sqlite3
```
Comment (asogaard) [inline examples/bjørn_low_weight_fitter_example.py:1]: I would rename this file to `examples/bjoern_low_weight_fitter_example.py`, see below.

Comment (RasmusOrsoe) [reply examples/bjørn_low_weight_fitter_example.py]: Done

**Comment by RasmusOrsoe** [comment]
@asogaard I believe everything is ready now

**Review by asogaard — APPROVED** [review]

**Pull request merged**